addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/TestInfoSessionIdURLUnitTest.java

```diff
@@ -0,0 +1,427 @@
+
+        // When
+        rule.scanHttpRequestSend(msg, -1);
+
+        // Then
+        assertEquals(0, alertsRaised.size());
+    }
+
+    @Test
+    public void noSessionIDAsUrlParameterInHTTPS()
+            throws HttpMalformedHeaderException, URIException {
+
+        // Given
+        String testURI = "https://example.com/session/foo?session=false";
+        HttpMessage msg = createHttpMessageWithRespBody(BODY);
+        msg.getRequestHeader().setURI(new URI(testURI, false));
+
+        // When
+        rule.scanHttpRequestSend(msg, -1);
+
+        // Then
+        assertEquals(0, alertsRaised.size());
+    }
+
+    @Test
+    public void containsJSESSIONIDAsUrlParameterInHTTPSOnCustomPort()
+            throws HttpMalformedHeaderException, URIException {
+
+        // Given
+        String testURI = "https://example.com:4443/foo?jsessionid=1a530637289b03x07199de8D531427";
+        HttpMessage msg = createHttpMessageWithRespBody(BODY);
+        msg.getRequestHeader().setURI(new URI(testURI, false));
+
+        // When
+
+        rule.scanHttpRequestSend(msg, -1);
+
+        // Then
+        assertEquals(1, alertsRaised.size());
+    }
+
+    @Test
+    public void containsJSESSIONIDInUrlBeforeParams()
+            throws HttpMalformedHeaderException, URIException {
+
+        // Given
+        String testURI = "http://tld.gtld/fred;JSESSIONID=asdfasdfasdf1234?foo=bar";
+        HttpMessage msg = createHttpMessageWithRespBody(BODY);
+        msg.getRequestHeader().setURI(new URI(testURI, false));
+
+        // When
+        rule.scanHttpRequestSend(msg, -1);
+
+        // Then
+        assertEquals(1, alertsRaised.size());
+    }
+
+    @Test
+    public void containsCFIDAsUrlParameter() throws HttpMalformedHeaderException, URIException {
+
+        // Given
+        String testURI = "http://example.com/foo?CFiD=1A530637289A03B07199A44E8D531427";
+        HttpMessage msg = createHttpMessageWithRespBody(BODY);
+        msg.getRequestHeader().setURI(new URI(testURI, false));
+
+        // When
+        rule.scanHttpRequestSend(msg, -1);
+
+        // Then
+        assertEquals(1, alertsRaised.size());
+    }
+
+    @Test
+    public void containsCFIDInUrlBeforeParams() throws HttpMalformedHeaderException, URIException {
+
+        // Given
+        String testURI = "http://tld.gtld/fred;CFID=asdfasdfasdf1234?foo=bar";
+        HttpMessage msg = createHttpMessageWithRespBody(BODY);
+        msg.getRequestHeader().setURI(new URI(testURI, false));
+
+        // When
+        rule.scanHttpRequestSend(msg, -1);
+
+        // Then
+        assertEquals(1, alertsRaised.size());
+    }
+
+    // @Test
+    // TODO: The scanHttpResponseReceive() method in this scanner currently does not
+    // look for session IDs in the response embedded in HREFs. As such, this test
+    // case fails so is commented out. When this scanner is enhanced to do this,
+    // enable this test case.
+    public void containsJSESSIONIDInResponseHREFParams()
```

Comment:
It can keep the `Test` annotation along with `Ignore` (which can have the reason). Same for following test.